### DISCUSSION TITLE
Unexpected behavior when using Task List with Inline Code.

### DISCUSSION THREAD

**Discussion by elitegoliath** [requarks/wiki#3288] [Error / Bug Report]
**Describe the bug**
When attempting to add text that has inline code to a task list, the preview renders it once in plain text, then creates a new unordered list item (no longer a task list item) under it with the properly formatted text. When viewing it on the rendered page the text renders like you'd expect, but still contains the extra line and it's an Unordered List item instead of a Task List item.

**To Reproduce**
1. Open a markdown editor for a page.
2. Create a task list entry using `- [ ] Some entry.`
2. Create another task list entry under the first one, this time with inline code: `- [ ] Here's some ``inline code`` `
3. Save.
4. Look at the preview to see the issue.
5. Close the editor.
6. See that the entry with inline code is an Unordered List Item.

**Expected behavior**
A Task List Item that contains inline code should remain a Task List Item while rendering inline code with the appropriate styles.

**Screenshots**
##### Issue displayed in the Preview.
![Preview Mode Issue](https://user-images.githubusercontent.com/12138681/103048128-2dec6000-4553-11eb-8b01-fbd699b9f0cb.png)

##### Issue displayed on the rendered page.
![Rendered Issue](https://user-images.githubusercontent.com/12138681/103048143-3775c800-4553-11eb-9d1e-f6318f7d2a88.png)

**Host Info (please complete the following information):**
 - OS: Docker Container (Linux)
 - Wiki.js version: Wiki.js 2.5.170
 - Database engine: PostgreSQL 11.7

**Additional context**
May or may not be relevant, but I have the Task List within a dropdown using the `<details>`.

**Comment by elitegoliath** [comment]
Same seems to happen when trying to use Links with it too. `[Some Link](http://thing.place)`

**Comment by book000** [comment]
I have found a problem related to this.  
Creating a task with code like the following will result in unexpected behavior:
```
- [x] `T`est
```
![](https://user-images.githubusercontent.com/8929706/103313582-d9ba1200-4a63-11eb-8c7e-6412633db5eb.png)
